kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/channel/Coordinator.java

```diff
@@ -218,6 +220,10 @@ private void commitToTable(
             .filter(distinctByKey(deleteFile -> deleteFile.path().toString()))
             .collect(Collectors.toList());
 
+    if (terminated) {
+      throw new ConnectException("Coordinator is terminated, commit aborted");
+    }
```

Comment:
I felt putting it as close as possible to the commit was best, so if `terminate` is called it will prevent the commit ASAP.

Reply:
yup, was mostly comming from the case where we entered the `commitToTable` block and the terminate was flipped, we could save on load table etc.., it seems like it will always be a race, make sense to check just before the commit.